hardhat.config.js

```diff
@@ -0,0 +1,164 @@
+require('dotenv').config()
+
+require('@1hive/hardhat-aragon')
+require('@nomiclabs/hardhat-ethers')
+require('@nomiclabs/hardhat-truffle5')
+require('@nomiclabs/hardhat-web3')
+require('hardhat-deploy')
+require('hardhat-gas-reporter')
+require('solidity-coverage')
+
+const { node_url, accounts } = require('./utils/network')
+
+process.removeAllListeners('warning')
+module.exports = {
+  solidity: {
+    compilers: [
+      {
+        version: '0.4.24',
+        settings: {
+          optimizer: {
+            enabled: true,
+            runs: 10000,
+          },
+        },
+      },
+    ],
+  },
+  aragon: {
+    appEnsName: 'augmented-bonding-curve.open.aragonpm.eth',
+    appContractName: 'AugmentedBondingCurve',
+    appRoles: [
+      {
+        name: 'Update beneficiary address',
+        id: 'UPDATE_FORMULA_ROLE',
+        params: [],
+      },
+      {
+        name: 'Update beneficiary address',
+        id: 'UPDATE_BENEFICIARY_ROLE',
+        params: [],
+      },
+      {
+        name: 'Update fees',
+        id: 'UPDATE_FEES_ROLE',
+        params: [],
+      },
+      {
+        name: 'Add tokens as whitelisted collaterals',
+        id: 'ADD_COLLATERAL_TOKEN_ROLE',
+        params: [],
+      },
+      {
+        name: 'Remove tokens as whitelisted collaterals',
+        id: 'REMOVE_COLLATERAL_TOKEN_ROLE',
+        params: [],
+      },
+      {
+        name: 'Update collateral tokens collateralization settings',
+        id: 'UPDATE_COLLATERAL_TOKEN_ROLE',
+        params: [],
+      },
+      {
+        name: 'Open trading',
+        id: 'OPEN_TRADING_ROLE',
+        params: [],
+      },
+      {
+        name: 'Make buy orders',
+        id: 'MAKE_BUY_ORDER_ROLE',
+        params: ['Buyer'],
+      },
+      {
+        name: 'Make sell orders',
+        id: 'MAKE_SELL_ORDER_ROLE',
+        params: ['Seller'],
+      },
+    ],
+    appBuildOutputPath: 'public/',
+  },
+  networks: {
+    hardhat: {
+      // process.env.HARDHAT_FORK will specify the network that the fork is made from.
+      // this line ensure the use of the corresponding accounts
+      accounts: accounts(process.env.HARDHAT_FORK),
+      forking: process.env.HARDHAT_FORK
+        ? {
+            url: node_url(process.env.HARDHAT_FORK),
+            blockNumber: process.env.HARDHAT_FORK_NUMBER
+              ? parseInt(process.env.HARDHAT_FORK_NUMBER)
+              : undefined,
+          }
+        : undefined,
+    },
+    localhost: {
+      url: node_url('localhost'),
+      accounts: accounts(),
+      ensRegistry: '0xaafca6b0c89521752e559650206d7c925fd0e530',
```

Comment:
The `ensRegistry` is also an extension configured by `hardhat-aragon` plugin. That allows us to set up a custom `ensRegistry` that we will be using when publishing our app to AragonPM.